src/Expression/AbstractArrayExpressionBuilder.php

```diff
@@ -68,13 +69,17 @@ public function __construct(protected readonly QueryBuilderInterface $queryBuild
     /**
      * The Method builds the raw SQL from the `$expression` that won't be additionally escaped or quoted.
      *
-     * @param ArrayExpression $expression The expression to build.
+     * @param ExpressionInterface $expression The expression to build.
      * @param array $params The binding parameters.
      *
      * @return string The raw SQL that won't be additionally escaped or quoted.
      */
     public function build(ExpressionInterface $expression, array &$params = []): string
     {
+        if (!$expression instanceof ArrayExpression) {
+            throw new InvalidArgumentException(static::class . ' could only be used with ArrayExpression.');
```

Comment:
Should we throw `InvalidArgumentException` or `InvalidConfigException` here?

Reply:
The type check seems redundant, the value is then passed to a strongly typed method `buildValue()` where the right type specified.